Automattic/newspack-newsletters
# feat/share block

### All Submissions:






### Changes proposed in this Pull Request:



Closes #349.

### How to test the changes in this Pull Request:

1. Create or edit a newsletter, insert the new "Share" block
2. If the newsletter is not marked as public, observe a notice prompting to make it so
3. Observe a "Forwarded email content" field in the editor sidebar, when editing the block
3. Send the newsletter - observe that after clicking on the block in the email client, a new email window appears
4. Observe the title of the new email is `Fwd: <newsletter title>`, and the content is the value of the "Forwarded email content" field, with `[LINK]` replaced by the link to the public newsletter

## Caveats

1. If there's a way to make a "forward this email" link in HTML email, that'd be a better solution
2. The block validation will fail, but only with an error in the JS console, which should not disrupt the editing experience. See 055ad5ec082de2735fa16543dcc7ae0279c15181 

### Other information:
